src/data/dfinsupp.lean

```diff
@@ -725,6 +772,21 @@ by { ext j, by_cases h1 : j = i; by_cases h2 : f j ≠ 0; simp at h2; simp [h1,
   (f.erase i).support = f.support.erase i :=
 by { ext j, by_cases h1 : j = i; by_cases h2 : f j ≠ 0; simp at h2; simp [h1, h2] }
 
+lemma support_update (f : Π₀ i, β i) (i : ι) (b : β i) [decidable (b = 0)] :
+  support (f.update i b) = if b = 0 then support (f.erase i) else insert i f.support :=
+begin
+  ext j,
+  split_ifs with hb,
+  { simp only [hb, update_eq_erase, support_erase] },
+  { rcases eq_or_ne i j with rfl|hi,
+    { simp [hb] },
+    { simp [hi.symm] } }
+end
+
+lemma support_update_ne_zero (f : Π₀ i, β i) (i : ι) {b : β i} [decidable (b = 0)] (h : b ≠ 0) :
+  support (f.update i b) = insert i f.support :=
+by rw [support_update, if_neg h]
```

Comment:
I think this would be tidier if you proved them in the other order; the second `{}` after the `split_ifs` is just this second lemma, right?